lib/inferno/dsl/fhir_evaluation/rules/all_resources_reachable.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require_relative '../../../utils/evaluator_util'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      module Rules
+        class AllResourcesReachable < Rule
+          attr_accessor :config
+
+          def check(context)
+            @config = context.config
+            # TODO: add some customizable configurations for this rule:
+            # - whether it should be checked at all
+            # - determine the base/root resource
+            # - which types/resources to ignore
+            # TODO: for the customizable configurations, use the config settings below:
+            # config.Rule.AllResourcesReachable.CheckedAtAll: true/false
+            # config.Rule.AllResourcesReachable.RunOnlyBaseResource: true/false
+            # config.Rule.AllResourcesReachable.IgnoreType: - value
+            # TODO: can come up with a "connectedness metric" to see how well-connected the data set is
+
+            # every resource is either making a resolvable reference or is referenced
+            util = Inferno::Utils::EvaluatorUtil
+            @referenced_resources = Set.new
```

Comment:
Use `attr_accessor`s rather than directly referring to instance variables. This should be addressed in the other files as well.

Reply:
Done https://github.com/inferno-framework/inferno-core/pull/583/commits/85a2bd47e8264f679551c9a16a4a0643f5c3726e

Reply:
This has not been done. There are still a lot of direct references to instance variables in the reference extractor.

Reply:
Neither an answer nor question, but I am trying to be careful of changing underlying data and function structure. When the evaluator util was created, I think the original developer's intention was to have one reference extractor instance to one IG, initiated at the beginning of IG import then reused while running the evaluator. I am not sure of it for now and inclined to make functions stateless, literally as "function": give me input and I will give you back output, unless there's strong reason why it should be stateful, e.g. handle very complicated reasoning, or have to be reused many time..

Reply:
Modified https://github.com/inferno-framework/inferno-core/pull/583/commits/1872d88922e4166ac5b5cb1c48dcf33df2e8b3b7